nixos/modules/services/networking/nomad.nix

```diff
@@ -51,15 +51,17 @@ in
 
       settings = mkOption {
         type = format.type;
-        default = { };
+        default = {
+          # Agrees with `StateDirectory = "nomad"` set below.
+          data_dir = "/var/lib/nomad";
```

Comment:
Does `data_dir` need to be configurable? If so, then we should make `StateDirectory` match, and not set `StateDirectory` in cases where its dirname isn't `/var/lib`

Reply:
See my [previous comment](https://github.com/NixOS/nixpkgs/pull/105739#discussion_r557689413).

Additional to this you should add an `assertion` corresponding to `dropPrivileges` - the only possibility for `data_dir` is under `/var/lib` if `DynamicUser` is used.